### DISCUSSION TITLE
Need help with nautobot services starting after ubuntu kernel security update

### DISCUSSION THREAD

**Discussion by ilouie** [nautobot/nautobot#4944] [General]
Sorry, not familiar with setup of Nautobot...except for using it. Just had to udpate Ubuntu kernel to address some security vulnerabilities, and now nautobot will not start.  Requesting guidance...

nautobot@it-nautobot:~$ systemctl status  nautobot nautobot-worker
● nautobot.service - Nautobot WSGI Service
     Loaded: loaded (/etc/systemd/system/nautobot.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Fri 2023-12-08 12:40:23 PST; 19s ago
       Docs: https://nautobot.readthedocs.io/en/stable/
    Process: 6589 ExecStart=/home/nautobot/bin/nautobot-server start --pidfile /var/tmp/nautobot.pid --ini /home/nautobot/uwsgi.ini (code=exited, status=1/FAILURE)
   Main PID: 6589 (code=exited, status=1/FAILURE)

● nautobot-worker.service - Nautobot Request Queue Worker
     Loaded: loaded (/etc/systemd/system/nautobot-worker.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Fri 2023-12-08 12:40:30 PST; 11s ago
       Docs: https://nautobot.readthedocs.io/en/stable/
    Process: 6596 ExecStart=/home/nautobot/bin/nautobot-server rqworker (code=exited, status=1/FAILURE)
   Main PID: 6596 (code=exited, status=1/FAILURE)
   
    sudo journalctl -xu nautobot-worker
    
Dec 08 13:03:40 it-nautobot systemd[1]: nautobot.service: Scheduled restart job, restart counter is at 21.
Dec 08 13:03:40 it-nautobot systemd[1]: Stopped Nautobot WSGI Service.
Dec 08 13:03:40 it-nautobot systemd[1]: Started Nautobot WSGI Service.
Dec 08 13:03:41 it-nautobot nautobot-server[4446]: Traceback (most recent call last):
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "/home/nautobot/lib/python3.8/site-packages/nautobot/extras/plugins/utils.py", line 63, in >
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:     plugin = importlib.import_module(plugin_name)
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:     return _bootstrap._gcd_import(name[level:], package, level)
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
Dec 08 13:03:41 it-nautobot nautobot-server[4446]: ModuleNotFoundError: No module named 'nautobot_ssot'
Dec 08 13:03:41 it-nautobot nautobot-server[4446]: The above exception was the direct cause of the following exception:
Dec 08 13:03:41 it-nautobot nautobot-server[4446]: Traceback (most recent call last):
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "/home/nautobot/lib/python3.8/site-packages/nautobot/core/runner/runner.py", line 118, in s>
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:     initializer(
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "/home/nautobot/lib/python3.8/site-packages/nautobot/core/cli.py", line 173, in _configure_>
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:     load_plugins(settings)
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "/home/nautobot/lib/python3.8/site-packages/nautobot/extras/plugins/utils.py", line 53, in >
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:     load_plugin(plugin_name, settings)
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:   File "/home/nautobot/lib/python3.8/site-packages/nautobot/extras/plugins/utils.py", line 66, in >
Dec 08 13:03:41 it-nautobot nautobot-server[4446]:     raise PluginNotFound(
Dec 08 13:03:41 it-nautobot nautobot-server[4446]: nautobot.extras.plugins.exceptions.PluginNotFound: Unable to import plugin nautobot_ssot: Module n>
Dec 08 13:03:41 it-nautobot systemd[1]: nautobot.service: Main process exited, code=exited, status=1/FAILURE
Dec 08 13:03:41 it-nautobot systemd[1]: nautobot.service: Failed with result 'exit-code'.

**Comment by itdependsnetworks** [comment]
This is saying that `nautobot_ssot` is not installed. This could be as simple as running `pip install nautobot-ssot`, depending on your environment. Keep in mind, you may continue to get similar messages with different plugins.. depending on your setup, and will require similar package installations. 

```
Dec 08 13:03:41 it-nautobot nautobot-server[4446]: nautobot.extras.plugins.exceptions.PluginNotFound: Unable to import plugin nautobot_ssot: Module n>
```

**Comment by ilouie** [comment]
I removed ssot and the config in nautobot_config.py, and restarted services.  The good news is that this nautobot server does not ssot. :) Services restarted after the removal of ssot.  Thank you so much for helping!
